src/image/prepare_single_image_build_matrix.py

```diff
@@ -228,7 +228,7 @@ def main():
         # so let's remove this field since we don't need it for the builds
         del build["release"]
 
-    release_to = "true" if "release" in image_trigger else ""
+    release_to = "true" if builds else ""
```

Comment:
I don't think this is the desired behavior. We don't release every time we build. You may want to build a rock without releasing it (thus staying just in GHCR)

Reply:
why not

```python
    release_to = "true" if "release" in image_trigger else ""

    for build in builds:
        write_revision_data(args.revision_data_dir, build)
        # the workflow GH matrix has a problem parsing nested JSON dicts
        # so let's remove this field since we don't need it for the builds
        if "release" in build:
            release_to = "true"
            del build["release"]
```

Reply:
Right! I see what you mean now.

Reply:
I just pushed some changes fix that. We also use a bool for release_to for clarity.